From: Andrew Morton <akpm@linux-foundation.org>
Subject: mm-memcontrol-eliminate-the-problem-of-dying-memory-cgroup-for-lru-folios-fix
Date: Thu Jan 15 12:37:55 PM PST 2026

add missed rcu_unlock(), needed by "mm/swap_cgroup: fix kernel BUG in
swap_cgroup_record".

Reported-by: "David Hildenbrand (Red Hat)" <david@kernel.org>
Reported-by: kernel test robot <lkp@intel.com>
Cc: Muchun Song <songmuchun@bytedance.com>
Cc: Qi Zheng <zhengqi.arch@bytedance.com>
Cc: Deepanshu Kartikey <kartikey406@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memcontrol-v1.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/mm/memcontrol-v1.c~mm-memcontrol-eliminate-the-problem-of-dying-memory-cgroup-for-lru-folios-fix
+++ a/mm/memcontrol-v1.c
@@ -639,8 +639,10 @@ void memcg1_swapout(struct folio *folio,
 	 * in swapcache, reusing the same swap entries.
 	 */
 	oldid = lookup_swap_cgroup_id(entry);
-	if (oldid == mem_cgroup_id(memcg))
+	if (oldid == mem_cgroup_id(memcg)) {
+		rcu_read_unlock();
 		return;
+	}
 	VM_WARN_ON_ONCE(oldid != 0);
 
 	/*
_
